tests/run/const_variable.pyx

```diff
@@ -0,0 +1,54 @@
+# mode: run
+
+cdef const int int_constant = 50
+cdef const int int_constant2 = int_constant
+cdef const int int_constant3 = 5
+cdef const int *pint_constant = &int_constant
+cdef const int int_sum_constant = 50 + 50
+# FIXME:
+# cdef const int int_sum_constant = 10 + int_constant
+
+cdef const float float_constant = 50.5
+# FIXME:
+# cdef const float float_sum_constant = 50.2 + 50.3
+# cdef const float float_sum_constant = 50.2 + float_constant
+
+
+def test_constant_int_value():
+    """
+    >>> test_constant_int_value()
+    50
+    50
+    100
+    """
+    print(int_constant)
+    print(int_constant2)
+    print(int_sum_constant)
+
+def test_array():
+    """
+    >>> test_array()
+    5
+    """
+    cdef int[int_constant3] int_array
```

Comment:
The other thing worth pointing out here: if you put it into Godbolt you see it's generating a C99 variable length array (which isn't great because it's not C89 compatible, and is also optional after C11).

On C++ it generates a fixed length array though.

Again, I think the solution involves generating enums, but it's obviously a bit fiddly...